src/Aether/Devices/Drivers/Sgp4x.cs

```diff
@@ -0,0 +1,162 @@
+        /// </summary>
+        /// <returns>The serial number of the device.</returns>
+        public ReadOnlySpan<byte> GetSerialNumber()
+        {
+            Span<byte> serialNumberWithCRC = stackalloc byte[9];
+
+            // Transmit command
+            _device.WriteByte(0x36);
+            _device.WriteByte(0x82);
+
+            Thread.Sleep(1);
+
+            // Read first two bytes of serial number (+ CRC)
+            Sensirion.ReadUInt16BigEndianAndCRC8(serialNumberWithCRC.Slice(0, 3));
+            
+            // Read second two bytes of serial number (+ CRC)
+            Sensirion.ReadUInt16BigEndianAndCRC8(serialNumberWithCRC.Slice(4, 3));
+            
+            // Read third two bytes of serial number (+ CRC)
+            Sensirion.ReadUInt16BigEndianAndCRC8(serialNumberWithCRC.Slice(7, 3));
+
+            // Read serial number into array excluding CRC bytes
+            byte[] serialNumber = new byte[6];
+
+            serialNumber[0] = serialNumberWithCRC[0];
+            serialNumber[1] = serialNumberWithCRC[1];
+            serialNumber[2] = serialNumberWithCRC[3];
+            serialNumber[3] = serialNumberWithCRC[4];
+            serialNumber[4] = serialNumberWithCRC[6];
+            serialNumber[5] = serialNumberWithCRC[7];
+
+            return new ReadOnlySpan<byte>(serialNumber);
+        }
+
+        /// <summary>
+        /// Runs the sensors self test.
+        /// </summary>
+        /// <returns>True if all tests passed. False if one or more tests failed.</returns>
+        public bool RunSelfTest()
+        {
+            Span<byte> testResultWithCRC = stackalloc byte[3];
+
+            // Transmit command
+            _device.WriteByte(0x28);
+            _device.WriteByte(0x0E);
+
+            Thread.Sleep(320);
+
+            // Read test result data
+            Sensirion.ReadUInt16BigEndianAndCRC8(testResultWithCRC);
+
+            // Check result status (Most significant byte, ignore least significant non-crc byte)
+            // 0xD4 = All tests passed
+            // 0x4B = One or more test failed
+            return testResultWithCRC[0] == 0xD4;
+        }
+
+        /// <summary>
+        /// Gets the raw VOC measurements from the sensor.
+        /// </summary>
+        /// <param name="relativeHumidityValue">The relative humidity value expressed as a percentage.</param>
+        /// <param name="temperatureValue">The temperature value expressed in degrees celsius.</param>
+        /// <param name="disableHotPlate">If true, the hot plate will be disabled after measurements are taken.</param>
+        /// <returns>The raw VOC measurement value from the sensor. If an error occurred, it will be <see langword="null"/>.</returns>
+        /// <remarks>If default values are supplied, humidity compensation will be disabled.</remarks>
+        public ushort? GetVOCRawMeasure(ushort relativeHumidityValue = 50, short temperatureValue = 25, bool disableHotPlate = true)
```

Comment:
The other sensor drivers are returning and taking UnitsNet types. For consistency, can we take those here instead of shorts?

Reply:
A `ushort` return isn't great either -- can the [VOC index algorithm](https://github.com/Sensirion/embedded-sgp/tree/master/sgp40_voc_index) be ported to C# (it looks like just the voc_algorithm files) here so we can return a proper VOC index?

Reply:
This is now using a custom unit called VolatileOrganicCompoundIndex